scripts/cfengine-build-host-setup.cf

```diff
@@ -0,0 +1,205 @@
+body file control
+{
+  inputs => { "$(sys.libdir)/stdlib.cf" };
+}
+# assumes that the operating system has working repo lists and has been updated/upgraded already
+
+bundle agent cfengine_build_host_setup
+{
+  packages:
+    debian_9|debian_10::
+      "python-psycopg2";
+    debian_11|debian_12::
+      "python3-psycopg2";
+    ubuntu_16:: # ubuntu_16 doesn't have systemd-coredump by default?
+      "systemd-coredump";
+    ubuntu_20::
+      "autoconf"; # because on arm ubu20 we need to reconfigure
+      "shellcheck"; # not sure why, ubu20 addition
+    debian.(!debian_12.!ubuntu_22)::
+      "python"; # debian_12 has only python3
+     !(debian_9|ubuntu_16).(debian|ubuntu)::
+       "default-jre"; # on debian10+ and ubuntu18+ this will be jdk11, good enough for jenkins 2.426.1 https://www.jenkins.io/doc/book/platform-information/support-policy-java/index.html
+
+# these packages were apt purge in shell scripts, package_method generic should do the same thing with package_method => "delete"
+    debian|ubuntu::
+      "libltdl7" package_policy => "delete";
+      "libltdl-dev" package_policy => "delete";
+
+# todo --allow-change-held-packages introduced in debian-12, why?
+      "binutils";
+      "bison";
+      "build-essential";
+      "curl"; # added for debian-10
+      "debhelper";
+      "dpkg-dev";
+      "expat";
+      "fakeroot";
+      "flex";
+      "gdb";
+      "libncurses5"; # added for debian-10
+      "libncurses5-dev"; # added for debian-10
+      "libexpat1-dev";
+      "libmodule-load-conditional-perl";
+      "libpam0g-dev";
+      "ntp";
+      "pkg-config";
+      "psmisc";
+      "python3-pip";
+      "rsync"; # added for debian-10
+
+    ubuntu_16.mingw_build_host:: # for now, only ubu16 hosts mingw builds
+      "wine:i386";
+      "mingw-w64";
+
+# I attempted to arrange these packages in order of: generic (all versions) and then as if we gradually added them through time: rhel-6, 7, 8, 9...
```

Comment:
On the second read through this it struk me that this could be a good job for `classfiltercsv()`.


```
# package, classexpression, comment
wine:i386, ubuntu_16.mingw_build_host,
mingw-w64, ubuntu_16.mingw_build_host,
expat-devel, redhat,
java-11-openjdk, redhat_6|centos_6|redhat_7|centos_7, OK for jenkins 2.426.1  https://www.jenkins.io/doc/book/platform-information/support-policy-java/index.html
```

Reply:
I shudder a bit at the complexity of it. I was leaning more towards a meta-package-name system where we clearly say what we want at a high level on ALL build hosts and then let that filter down to specifics on each platform.

Also I like the idea of keeping the reason and the collection more "together" which could also be done with multiple csv files and comments. Will keep it in mind for next go-around. :+1: Very nice. 🙏